### PR TITLE
[JENKINS-54128] Deprecated calls to Run#getLogFile

### PR DESCRIPTION
**Pull request by basil**
See [JENKINS-54128](https://issues.jenkins-ci.org/browse/JENKINS-54128). Eliminates the last call to `Run#getLogFile` by completely deleting the `LogFileReader` class (and `LogFileReaderTest`), along with its associated class `LineCountSupplier`. In its place, I am now using `new BufferedReader(build.getLogReader())`, which provides the same abstraction of iterating over log files lines. This change is now possible due to the previous changes I have been making which moved any non-essential functionality out of `LogFileReader` and into other classes. With `LogFileReader` gone, there wasn't much of a point to having separate classes for `FreestyleTimestampLogFileLineAccessor` and `PipelineTimestampLogFileLineAccessor`, so I combined them into a single `TimestampLogFileLineAccessor` class that covers all cases. There are three main cases, all of which are covered by integration tests:

1. Freestyle builds with a timestamp log file (covered by `TimestamperIntegrationTest#timestamperApi`)
2. Freestyle builds with `TimestampNote`s or Pipeline builds prior to version 1.9 (covered by `TimestamperIntegrationTest#timestampNote`)
3. Pipeline builds for version 1.9 or later (covered by `PipelineTest#timestamperApi`)

I ran these integration tests with JaCoCo and ensured we were getting full coverage of both `TimestampsActionOutput` and `TimestampLogFileLineAccessor`. Since the old mock-based `TimestampsActionOutputTest` was now failing and `TimestampsActionOutput` had full coverage with the new integration tests, I chose to simply delete `TimestampsActionOutputTest` rather than try and write new mocks. The integration tests provides more realistic testing anyhow.

With this change, the `consoleText` URL now works properly for Pipeline builds (I tested this manually), so I was also able to delete the remaining TODO from `usersettings.jelly`.

### CODE REVIEW

File: src/main/java/hudson/plugins/timestamper/io/LogFileReader.java [hunk 1]
```diff
@@ -1,108 +0,0 @@
...
-    this.build = checkNotNull(build);
-  }
-
-  /**
-   * Read the next line from the log file.
-   *
-   * @return the next line, or {@link Optional#empty()} if there are no more to read
-   */
-  public Optional<String> nextLine() throws IOException {
-    if (!build.getLogFile().exists()) { // TODO JENKINS-54128 rather use getLogText
```
Comment (jglick) [inline src/main/java/hudson/plugins/timestamper/io/LogFileReader.java]: FTR here

File: src/main/java/hudson/plugins/timestamper/io/LogFileReader.java [hunk 2]
```diff
@@ -1,108 +0,0 @@
...
-  }
-
-  /**
-   * Get the number of lines that can be read from the log file.
-   *
-   * @return the line count
-   */
-  @SuppressFBWarnings("RV_DONT_JUST_NULL_CHECK_READLINE")
-  public int lineCount() throws IOException {
-    if (!build.getLogFile().exists()) {
```
Comment (jglick) [inline src/main/java/hudson/plugins/timestamper/io/LogFileReader.java]: and here

**Comment by jglick** [comment]
I was about to run a sanity check against `pipeline-cloudwatch-logs` (basically to confirm that this patch fixes some exception there) until I remembered https://github.com/jenkinsci/pipeline-cloudwatch-logs-plugin/blob/5c0ddecd24882f06c35abdd2cee2aff0ccbc8dd7/src/main/java/io/jenkins/plugins/pipeline_cloudwatch_logs/CloudWatchRetriever.java#L222 that it would be better to have it emit `timestamper`-compatible output from the timestamps it already has.

**Comment by basil** [comment]
> Do not know enough about the code to comment in detail on the diff, but the description sounds right.

Thanks for taking a look!

**Comment by basil** [comment]
> I was about to run a sanity check against `pipeline-cloudwatch-logs` (basically to confirm that this patch fixes some exception there) until I remembered https://github.com/jenkinsci/pipeline-cloudwatch-logs-plugin/blob/5c0ddecd24882f06c35abdd2cee2aff0ccbc8dd7/src/main/java/io/jenkins/plugins/pipeline_cloudwatch_logs/CloudWatchRetriever.java#L222 that it would be better to have it emit `timestamper`-compatible output from the timestamps it already has.

Indeed. As long as it emits timestamps using either the Pipeline native format from `GlobalDecorator` or `TimestampNote`s, this logic should be able to parse it.

**Pull request merged**